### PR TITLE
Cookies

### PR DESCRIPTION
**Pull request by aidan-lane**
### Description of change(s), including why:

Added login cookie. This automatically logs in the user if they have logged in previously.

### Type of change(s)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Any work to still do or other concerns:

- May be necessary to add more values to store such as username/email etc. Should probably also be given the option to choose whether or not a cookie is saved.

### Closing issues:

- N/A

### CODE REVIEW

**Comment by aidan-lane** [comment]
I pulled from the "login" branch, so they PR doesn't have to be merged if this one is.

**Review by hbh7 — CHANGES_REQUESTED** [review]
The only concern I have is where cookies should be handled. To my knowledge, express-session in the backend is already creating and sending a cookie. Another probably isn't needed in the frontend. The utility to access it could definitely be useful though as I *think* our best bet is to have the backend tack on the username, name, etc. to the already being sent cookie and then React can read that in rather than use State for it (I think State is lost whenever the page is closed, so a cookie would definitely be better than State).

So, in simpler terms, check what the backend is doing with cookies already and see if this is redundant or not. I think at most we'll want to do is have React not make the cookie and use the existing one (if present) and everything else just the slight tweaks to enable that.

**Comment by aidan-lane** [comment]
switched to new branch -> login-session
